validator/src/journal/chain_head_lock.rs

```diff
@@ -0,0 +1,54 @@
+use std::sync::RwLockWriteGuard;
+
+use batch::Batch;
+use journal::block_wrapper::BlockWrapper;
+use journal::publisher::{BlockPublisherImpl, BlockPublisherState};
+
+/// Abstracts acquiring the lock used by the BlockPublisher without exposing access to the
+/// publisher itself.
+#[derive(Clone)]
+pub struct ChainHeadLock {
+    publisher: BlockPublisherImpl,
+}
+
+impl ChainHeadLock {
+    pub fn new(publisher: BlockPublisherImpl) -> Self {
+        ChainHeadLock { publisher }
+    }
+
+    pub extern "C" fn acquire(&self) -> ChainHeadGuard {
```

Comment:
Does this need to be extern "C"?

Reply:
I don't think so.

Reply:
We've been pretty consistent about using that, though I think it has become optional in recent versions of Rust

Reply:
In other words, if we want to get rid of it, it should be across the board (i.e. in a separate PR)

Reply:
Oh, wait, nevermind, I mis-read where this was being commented on.

Reply:
:)